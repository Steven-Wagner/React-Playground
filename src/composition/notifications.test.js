import React from 'react';
import ReactDOM from 'react-dom';
import Notifications from './notifications';
import renderer from 'react-test-renderer';


describe('Notifications Component', () => {
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Notifications />, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<Notifications name="message" unread={4}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<Notifications name="message" unread={0}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})