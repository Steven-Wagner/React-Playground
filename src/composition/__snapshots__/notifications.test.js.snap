// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Component renders the UI as expected 1`] = `
<div
  className="message"
>
  message
  <div
    className="number badge example"
  >
    4
  </div>
</div>
`;

exports[`Notifications Component renders the UI as expected 2`] = `
<div
  className="message"
>
  message
</div>
`;
